modules/trainer/GenericTrainer.py

```diff
@@ -319,6 +322,63 @@ def __sample_during_training(
 
         torch_gc()
 
+    def __validate(self, train_progress):
+        self.validation_data_loader.get_data_set().start_next_epoch()
```

Comment:
Hmm, question here. Should we only start an epoch if we're actually going to validate? That way we'll do a linear pass through variations, and guarantee to cycle through variations each time. If we do it this way, we aren't going to be cycling through variations in order (and if you're unlucky you're going to hit the same variation every time).

Reply:
i don't have any thoughts on that)